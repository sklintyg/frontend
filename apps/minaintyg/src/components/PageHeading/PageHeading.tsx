import { classNames } from '@frontend/components'
import { IDSIconPlus } from '@frontend/ids-react-ts'
import { ReactNode, useState } from 'react'

export function PageHeading({ heading, children }: { heading: string; children: ReactNode }) {
  const [clamped, setClamped] = useState(true)
  return (
    <div className="ids-content mb-7">
      <h1 className="ids-heading-1">{heading}</h1>
      <p data-clamped={clamped} className={classNames('ids-preamble mb-2', clamped && 'line-clamp-5 md:line-clamp-none')}>
        {children}
      </p>
      {clamped && (
        <button
          aria-label="Visa mer text"
          type="button"
          onClick={() => setClamped(!clamped)}
          className="text-sky-base flex items-center gap-2 md:hidden"
        >
          <IDSIconPlus width="100%" height="100%" inline className="bg-sky-base h-5 w-5 rounded-full p-1 text-white" color="currentColor" />
          <span className="underline">Visa mer text</span>
        </button>
      )}
    </div>
  )
}
