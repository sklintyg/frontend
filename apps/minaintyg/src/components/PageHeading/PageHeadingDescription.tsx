import { ReactNode } from 'react'

export function PageHeadingDescription({ children }: { children: ReactNode }) {
  return <div className="ids-preamble mb-2 text-lg md:text-[1.375rem]/[2.25rem]">{children}</div>
}
