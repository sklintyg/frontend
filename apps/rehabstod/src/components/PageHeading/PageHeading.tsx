export function PageHeading({ title, subTitle }: { title: string; subTitle?: string }) {
  return (
    <>
      <h1 className="ids-heading-1 ids-small">{title}</h1>
      {subTitle && <h2 className="ids-heading-3 mb-10">{subTitle}</h2>}
    </>
  )
}
