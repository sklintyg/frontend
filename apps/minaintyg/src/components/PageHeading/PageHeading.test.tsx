import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { PageHeading } from './PageHeading'

it('should remove clamped when show more button is pressed', async () => {
  render(<PageHeading heading="Test">Lorem ipsum</PageHeading>)
  expect(screen.getByText('Lorem ipsum')).toHaveAttribute('data-clamped', 'true')
  await userEvent.click(screen.getByRole('button', { name: 'Visa mer text' }))
  expect(screen.getByText('Lorem ipsum')).toHaveAttribute('data-clamped', 'false')
})
