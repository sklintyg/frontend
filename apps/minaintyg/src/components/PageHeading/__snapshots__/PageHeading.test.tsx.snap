// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render correctly 1`] = `
<div>
  <header
    class="ids-content mb-7"
  >
    <h1
      class="ids-heading-1 text-[min(2.25rem,9.25vw)]/[min(2.25rem,9.25vw)] md:text-5xl"
    >
      Test
    </h1>
    <div
      class="ids-preamble mb-2 text-lg md:text-[1.375rem]/[2.25rem]"
    >
      Lorem ipsum
    </div>
  </header>
</div>
`;
